server/backend/ios_notifications/ios_device/iosDeviceService.go

```diff
@@ -0,0 +1,63 @@
+// Package ios_device provides functions to register and remove ios devices
+package ios_device
+
+import (
+	pb "github.com/TUM-Dev/Campus-Backend/server/api"
+	"github.com/TUM-Dev/Campus-Backend/server/backend/ios_notifications"
+	"github.com/TUM-Dev/Campus-Backend/server/model"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type Service struct {
+	Repository *Repository
+}
+
+var (
+	ErrCouldNotRegisterDevice = status.Error(codes.Internal, "Could not register device")
+	ErrCouldNotRemoveDevice   = status.Error(codes.Internal, "Could not remove device")
+)
+
+func (service *Service) RegisterDevice(request *pb.RegisterIOSDeviceRequest) (*pb.RegisterIOSDeviceReply, error) {
+	if err := ios_notifications.ValidateRegisterDevice(request); err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	device := model.IOSDevice{
+		DeviceID:  request.GetDeviceId(),
+		PublicKey: request.GetPublicKey(),
+	}
+
+	err := service.Repository.RegisterDevice(&device)
+
+	if err != nil {
+		return nil, ErrCouldNotRegisterDevice
+	}
+
+	return &pb.RegisterIOSDeviceReply{
+		DeviceId: device.DeviceID,
+	}, nil
+}
+
+func (service *Service) RemoveDevice(request *pb.RemoveIOSDeviceRequest) (*pb.RemoveIOSDeviceReply, error) {
+
+	if err := ios_notifications.ValidateRemoveDevice(request); err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	err := service.Repository.RemoveDevice(request.GetDeviceId())
+
+	if err != nil {
+		return nil, ErrCouldNotRemoveDevice
+	}
+
+	return &pb.RemoveIOSDeviceReply{
+		Message: "Successfully removed device",
```

Comment:
Parsing text is always hard especially in case you want to translate it. Better would be to either return JSON, or just a code like I have done it here: https://github.com/UWPX/UWPX-Push-Server#success-server---client

Reply:
Hmmm... Ich sehe den Punkt total. Allerdings würde ein Error Code ja bereits über gRPC oder http zurückgegeben werden?
Translations sind natürlich nochmal etwas anderes.

Momentan nutze ich den Endpoint in iOS garnicht. Ich würde ihn anpassen wenn ich ihn dann auch nutze und weiß wie ich es auf iOS Seite handle.

Reply:
OK, this sounds reasonable in case we use proper response codes for everything since then it is parsable again :)